.gemini/styleguide.md

```diff
@@ -0,0 +1,715 @@
+    """Convert HuggingFace config to KerasHub config."""
+    return {
+        "vocabulary_size": transformers_config["vocab_size"],
+        "num_layers": transformers_config["num_hidden_layers"],
+        "num_heads": transformers_config["num_attention_heads"],
+        "hidden_dim": transformers_config["hidden_size"],
+        "intermediate_dim": transformers_config["intermediate_size"],
+    }
+
+
+def convert_weights(backbone, loader, transformers_config):
+    """Convert HuggingFace weights to KerasHub weights."""
+    # Embedding layer
+    loader.port_weight(
+        keras_variable=backbone.get_layer("token_embedding").embeddings,
+        hf_weight_key="model.embed_tokens.weight",
+    )
+    
+    # Transformer layers
+    for i in range(backbone.num_layers):
+        layer = backbone.get_layer(f"transformer_layer_{i}")
+        hf_prefix = f"model.layers.{i}"
+        
+        # Attention weights
+        loader.port_weight(
+            keras_variable=layer.attention.query_dense.kernel,
+            hf_weight_key=f"{hf_prefix}.self_attn.q_proj.weight",
+            hook_fn=lambda hf_tensor, keras_shape: np.transpose(
+                np.reshape(hf_tensor, (keras_shape[0], keras_shape[2], keras_shape[1])),
+                axes=(0, 2, 1),
+            ),
+        )
+        # ... additional weight mappings
+```
+
+### Converter Test Requirements
+- Test that the converter can load HuggingFace models correctly
+- Verify class detection works for both backbone and task models
+- Test with `load_weights=False` to ensure config conversion works
+- Include numerical equivalence tests when possible
+
+### Example Converter Test
+```python
+"""Tests for MyModel HuggingFace converter."""
+
+import pytest
+
+from keras_hub.src.models.backbone import Backbone
+from keras_hub.src.models.my_model.my_model_backbone import MyModelBackbone
+from keras_hub.src.models.my_model.my_model_text_classifier import MyModelTextClassifier
+from keras_hub.src.models.text_classifier import TextClassifier
+from keras_hub.src.tests.test_case import TestCase
+
+
+class TestMyModelConverter(TestCase):
+    @pytest.mark.large
+    def test_convert_preset(self):
+        model = MyModelTextClassifier.from_preset(
+            "hf://huggingface/my-model-base", num_classes=2
+        )
+        prompt = "This is a test sentence."
+        model.predict([prompt])
+
+    @pytest.mark.large
+    def test_class_detection(self):
+        model = TextClassifier.from_preset(
+            "hf://huggingface/my-model-base",
+            num_classes=2,
+            load_weights=False,
+        )
+        self.assertIsInstance(model, MyModelTextClassifier)
+        
+        model = Backbone.from_preset(
+            "hf://huggingface/my-model-base",
+            load_weights=False,
+        )
+        self.assertIsInstance(model, MyModelBackbone)
+```
+
+## Code Quality Standards
+
+### Formatting
+- Use `black` for code formatting
```

Comment:
Hmmm, this should be ruff? Maybe, we should copy stuff from `CONTRIBUTING.md`.

Reply:
Can we also add an instruction: "If code format test fails, add a comment on how to run formatting and linting"? Does GCA work that way?